### PR TITLE
Update versioning.md

### PR DESCRIPTION
**Pull request by jtattermusch**
Addresses part of the concerns raised here: https://github.com/grpc/grpc/pull/19667#issuecomment-546408329

### CODE REVIEW

File: doc/versioning.md [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
+All gRPC implementations use a three-part version number (`vX.Y.Z`) and follow [semantic versioning](https://semver.org/), which defines the semantics of major, minor and patch components of the version number. In addition to that, gRPC versions evolve according to these rules:
 - **Major version bumps** only happen on rare occasions. In order to qualify for a major version bump, certain criteria described later in this document need to be met. Most importantly, a major version increase must not break wire compatibility with other gRPC implementations so that existing gRPC libraries remain fully interoperable. 
 - **Minor version bumps** happen approx. every 6 weeks as part of the normal release cycle as defined by the gRPC release process. A new release branch named vMAJOR.MINOR.PATCH) is cut every 6 weeks based on the [release schedule](https://github.com/grpc/grpc/blob/master/doc/grpc_release_schedule.md).
 - **Patch version bump** corresponds to bugfixes done on release branch.
 
+There a few situations where we don't adhere to the Semantic Versioning 2.0.0 strictly:
```
Comment (srini100) [inline doc/versioning.md]: Nit: There *are* a few....

File: doc/versioning.md [hunk 2]
```diff
@@ -2,11 +2,15 @@
...
+- A **minor** version will not necessarily add new functionality. This follows from the fact that we cut minor releases on a regular schedule, so we can't guarantee there's always going to be some new feature in each of the supported languages.
+- Backward compatibility can be broken by a **minor** release if the API affected by the change was marked as EXPERIMENTAL upon its introduction
```
Comment (srini100) [inline doc/versioning.md]: Nit: Full stop at the end.

File: doc/versioning.md [hunk 3]
```diff
@@ -2,11 +2,15 @@
...
 - **Major version bumps** only happen on rare occasions. In order to qualify for a major version bump, certain criteria described later in this document need to be met. Most importantly, a major version increase must not break wire compatibility with other gRPC implementations so that existing gRPC libraries remain fully interoperable. 
 - **Minor version bumps** happen approx. every 6 weeks as part of the normal release cycle as defined by the gRPC release process. A new release branch named vMAJOR.MINOR.PATCH) is cut every 6 weeks based on the [release schedule](https://github.com/grpc/grpc/blob/master/doc/grpc_release_schedule.md).
 - **Patch version bump** corresponds to bugfixes done on release branch.
 
+There a few situations where we don't adhere to the Semantic Versioning 2.0.0 strictly:
+- A **minor** version will not necessarily add new functionality. This follows from the fact that we cut minor releases on a regular schedule, so we can't guarantee there's always going to be some new feature in each of the supported languages.
```
Comment (dfawley) [inline doc/versioning.md]: Nit: "..., so we can't guarantee there will always be new features in ..."

**Comment by jtattermusch** [comment]
Docs only change, merging.

**Pull request merged**